packages/contracts-bedrock/src/L1/L2OutputOracle.sol

```diff
@@ -64,86 +60,82 @@ contract L2OutputOracle is Initializable, ISemver {
     event OutputsDeleted(uint256 indexed prevNextOutputIndex, uint256 indexed newNextOutputIndex);
 
     /// @notice Semantic version.
-    /// @custom:semver 1.6.0
-    string public constant version = "1.6.0";
+    /// @custom:semver 1.7.0
+    string public constant version = "1.7.0";
 
     /// @notice Constructs the L2OutputOracle contract.
     /// @param _submissionInterval  Interval in blocks at which checkpoints must be submitted.
     /// @param _l2BlockTime         The time per L2 block, in seconds.
-    /// @param _finalizationPeriodSeconds The amount of time that must pass for an output proposal
-    //                                    to be considered canonical.
-    constructor(uint256 _submissionInterval, uint256 _l2BlockTime, uint256 _finalizationPeriodSeconds) {
+    /// @param _startingBlockNumber The number of the first L2 block.
+    /// @param _startingTimestamp   The timestamp of the first L2 block.
+    /// @param _proposer            The address of the proposer.
+    /// @param _challenger          The address of the challenger.
+    constructor(
+        uint256 _submissionInterval,
+        uint256 _l2BlockTime,
+        uint256 _startingBlockNumber,
+        uint256 _startingTimestamp,
+        address _proposer,
+        address _challenger,
+        uint256 _finalizationPeriodSeconds
+    ) {
         require(_l2BlockTime > 0, "L2OutputOracle: L2 block time must be greater than 0");
         require(_submissionInterval > 0, "L2OutputOracle: submission interval must be greater than 0");
 
         SUBMISSION_INTERVAL = _submissionInterval;
         L2_BLOCK_TIME = _l2BlockTime;
+        PROPOSER = _proposer;
+        CHALLENGER = _challenger;
         FINALIZATION_PERIOD_SECONDS = _finalizationPeriodSeconds;
 
-        initialize({ _startingBlockNumber: 0, _startingTimestamp: 0, _proposer: address(0), _challenger: address(0) });
+        initialize(_startingBlockNumber, _startingTimestamp);
     }
 
     /// @notice Initializer.
     /// @param _startingBlockNumber Block number for the first recoded L2 block.
     /// @param _startingTimestamp   Timestamp for the first recoded L2 block.
-    /// @param _proposer            The address of the proposer.
-    /// @param _challenger          The address of the challenger.
-    function initialize(
-        uint256 _startingBlockNumber,
-        uint256 _startingTimestamp,
-        address _proposer,
-        address _challenger
-    )
-        public
-        reinitializer(Constants.INITIALIZER)
-    {
+    function initialize(uint256 _startingBlockNumber, uint256 _startingTimestamp) public initializer {
```

Comment:
I notice that we are going back from `reinitialize` to `initializer` - this will not work in practice unless during the 2 step upgrade we zero out the `_initialized` storage slot. This will add extra complexity to the upgrade step but will simplify the maintenance of the code. I lean towards adding complexity to the upgrade step so that we can more easily move between contract versions - ie we do not need to `reinitialize(prev value + 1)` on every version of semver bump to ensure that each contract is deployable

Reply:
I agree on that direction.